### PR TITLE
fix: approval disabled 

### PR DESCRIPTION
**Pull request by Mi-Lan**

### CODE REVIEW

**Comment by Mi-Lan** [comment]
https://user-images.githubusercontent.com/33226956/183256209-dd5319e0-4e21-4b1e-a82e-6bd8c6335ba3.mov

- I also think we should more flow with approval and fetch it in batch if we don't wanna see this blink happening when switching from intitial trade which is fetched first, as you can see for split second you see button to trade with cow and then it changes to approvals process. Thats because approval info wasn't fetched...

https://user-images.githubusercontent.com/33226956/183256307-645e15e1-2e3d-4ff9-bd45-570b3b38cd13.mov

**Comment by niemam29** [comment]
Tested and looks good to me!

**Comment by berteotti** [comment]
Got this error while testing. Approved SWPR on CoW and Swapr. Tried to swap in Swapr and didn't work. But Swapr swap is working on develop branch.

image

File: src/pages/Swap/index.tsx [hunk 1]
```diff
@@ -196,14 +197,27 @@ export default function Swap() {
...
   useEffect(() => {
     if (approval === ApprovalState.PENDING) {
-      setApprovalSubmitted(true)
+      const newArray = [...approvalsSubmitted]
+      newArray[currentTradeIndex] = true
+      setApprovalsSubmitted(newArray)
     }
-  }, [approval, approvalSubmitted])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [approval])
```
Comment (berteotti) [inline src/pages/Swap/index.tsx:220]: Doesn't this work with `approvalSubmitted`?

**Review by berteotti — COMMENTED** [review]
Looking good, just made 2 comments

File: src/pages/Swap/index.tsx [hunk 2]
```diff
@@ -196,14 +197,27 @@ export default function Swap() {
...
 
   // check if user has gone through approval process, used to show two step buttons, reset on token change
-  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  const [approvalsSubmitted, setApprovalsSubmitted] = useState<boolean[]>([])
+
+  const currentTradeIndex = useMemo(() => {
+    if (allPlatformTrades) {
+      return allPlatformTrades.findIndex(
+        trade => trade && platformOverride && trade.platform.name === platformOverride.name
+      )
```
Comment (adamazad) [inline src/pages/Swap/index.tsx:206]: `currentTradeIndex` can be -1
- [`findIndex` returns -1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) when no value is foudn.
- `trade` is always defined. 
- `platformOverride` is undefined by default.

File: src/pages/Swap/index.tsx (hunk 1 above)
Comment (Mi-Lan) [reply src/pages/Swap/index.tsx:220]: well its just reliant on the approval change rest are not neccesacry and we can cause unwanted re renders when we change approvals submitted and have it as dependency at the same time

File: src/pages/Swap/index.tsx (hunk 2 above)
Comment (Mi-Lan) [reply src/pages/Swap/index.tsx:206]: not sure what you are suggesting here?

Comment (adamazad) [reply src/pages/Swap/index.tsx:206]: It should return 0 in both cases

File: src/pages/Swap/index.tsx [hunk 3]
```diff
@@ -196,14 +197,28 @@ export default function Swap() {
...
   // check if user has gone through approval process, used to show two step buttons, reset on token change
-  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  const [approvalsSubmitted, setApprovalsSubmitted] = useState<boolean[]>([])
+
+  const currentTradeIndex = useMemo(() => {
+    if (allPlatformTrades) {
+      const index = allPlatformTrades.findIndex(
+        trade => trade && platformOverride && trade.platform.name === platformOverride.name
+      )
+      return index === -1 ? 0 : index
```
Comment (berteotti) [inline src/pages/Swap/index.tsx:207]: I think we can simplify here :)

```suggestion
     return allPlatformTrades.findIndex(
        trade => trade && platformOverride && trade.platform.name === platformOverride.name
      ) || 0
```

Comment (Mi-Lan) [reply src/pages/Swap/index.tsx]: thats much smoother thank you!

**Review by adamazad — APPROVED** [review]

**Review by berteotti — APPROVED** [review]

**Pull request merged**